### ISSUE TITLE
Don't switch on cancellation

### ISSUE THREAD

**Issue by summersz** [Kristijan/tmux-fzf-pane-switch#2]
Hi, thanks for creating this plugin. I love how simple it is for switching.

One gripe I have is that if you open the switcher and then decide to close it, it will open whichever pane is at the top of the list.

**Comment by Kristijan** [comment]
Hey @summersz, glad you like it.

I can replicate the problem, and it's been bugging me for a while, but not enough to do something about it. Now that I know someone else is impacted, let me take a look and see what I can do :)

**Comment by summersz** [comment]
I added a current pane check.  This code is also hiding the % prefix from fzf.

```
function select_pane() {
    local pane
    local pane_id
    local current_pane

    # Save the currently active pane ID
    current_pane=$(tmux display-message -p '#{pane_id}')

    # Check if we're using the fzf preview pane
    if [[ "${1}" = 'true' ]]; then
        pane=$(tmux list-panes -aF "${4}" |
            awk '{$1=substr($1, 2); print}' | 
           fzf --exit-0 --bind=enter:replace-query+print-query --reverse --tmux "${2}" --preview='tmux capture-pane -ep -t %{1}' --preview-label='pane preview' --preview-window="${3}")
    else
        pane=$(tmux list-panes -aF "${4}" | 
            awk '{$1=substr($1, 2); print}' | 
           fzf --exit-0 --bind=enter:replace-query+print-query --reverse --tmux "${2}")
    fi

    # Set pane_id to first part of fzf output
    pane_id=$(echo "${pane}" | awk '{print $1}')

    # If pane_id is empty, exit without changing pane
    if [[ -z "${pane_id}" ]]; then
        tmux switch-client -t "${current_pane}"
    # Check if pane exists
    elif tmux has-session -t "%${pane_id}" >/dev/null 2>&1; then
        # Found it! Let's switch.
        tmux switch-client -t "%${pane_id}"
    else
        # Pane not found, let's create it.
        tmux command-prompt -b -p "Press ENTER to create a new window in the current session [${pane}]" "new-window -n \"${pane}\""
    fi
}
```

**Comment by Kristijan** [comment]
> I added a current pane check. This code is also hiding the % prefix from fzf.
> 
> ```
> function select_pane() {
>     local pane
>     local pane_id
>     local current_pane
> 
>     # Save the currently active pane ID
>     current_pane=$(tmux display-message -p '#{pane_id}')
> 
>     # Check if we're using the fzf preview pane
>     if [[ "${1}" = 'true' ]]; then
>         pane=$(tmux list-panes -aF "${4}" |
>             awk '{$1=substr($1, 2); print}' | 
>            fzf --exit-0 --bind=enter:replace-query+print-query --reverse --tmux "${2}" --preview='tmux capture-pane -ep -t %{1}' --preview-label='pane preview' --preview-window="${3}")
>     else
>         pane=$(tmux list-panes -aF "${4}" | 
>             awk '{$1=substr($1, 2); print}' | 
>            fzf --exit-0 --bind=enter:replace-query+print-query --reverse --tmux "${2}")
>     fi
> 
>     # Set pane_id to first part of fzf output
>     pane_id=$(echo "${pane}" | awk '{print $1}')
> 
>     # If pane_id is empty, exit without changing pane
>     if [[ -z "${pane_id}" ]]; then
>         tmux switch-client -t "${current_pane}"
>     # Check if pane exists
>     elif tmux has-session -t "%${pane_id}" >/dev/null 2>&1; then
>         # Found it! Let's switch.
>         tmux switch-client -t "%${pane_id}"
>     else
>         # Pane not found, let's create it.
>         tmux command-prompt -b -p "Press ENTER to create a new window in the current session [${pane}]" "new-window -n \"${pane}\""
>     fi
> }
> ```

This works for me, thank-you! Did you want to raise a pull request with your changes, or you happy for me to just integrate them in and make another release?

**Comment by summersz** [comment]
Hi, happy for you to integrate them in, thanks!

**Comment by Kristijan** [comment]
Resolved by #3
